tests/Lewee.Application.Tests.Unit/PipelineBehaviorTests.cs

```diff
@@ -0,0 +1,213 @@
+using System.Net;
+using System.Net.Http.Json;
+using FluentAssertions;
+using Lewee.Application.Mediation.Requests;
+using Lewee.Shared;
+using Microsoft.Extensions.Logging;
+using Xunit;
+
+namespace Lewee.Application.Tests.Unit;
+
+/// <summary>
+/// Tests for pipeline behaviors using TestHost
+/// </summary>
+public class PipelineBehaviorTests : IClassFixture<TestWebApplicationFactory>
+{
+    private readonly TestWebApplicationFactory factory;
+    private readonly HttpClient client;
+
+    public PipelineBehaviorTests(TestWebApplicationFactory factory)
+    {
+        this.factory = factory;
```

Comment:
Use [TestServer](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.testhost.testserver?view=aspnetcore-9.0) instead of `WebApplicationFactory`.

This way you can test each behaviour in it's own test class as opposed testing all of them together.

Thesre are more partial integration tests than unit tests.